File: src/main/java/com/emarket/service/interfaz/ClienteService.java
package com.emarket.service.interfaz;

import com.emarket.dto.cliente.ClienteRequestDto;
import com.emarket.dto.cliente.ClienteResponseDto;

import java.util.List;

public interface ClienteService {
    ClienteResponseDto registrar(ClienteRequestDto dto);
    List<ClienteResponseDto> listar(String token);
    ClienteResponseDto eliminar(Long id);
}

File: frontend/src/services/cliente.service.ts
import { apiClient } from '../core/api/axiosInstance';
import type {
  ClienteRequestDto,
  ClienteResponseDto,
  ApiResponseDto,
} from '../core/types';

export const clienteService = {
  async registrar(dto: ClienteRequestDto): Promise<ClienteResponseDto> {
    const { data } = await apiClient.post<ClienteResponseDto>('/clientes/registro', dto);
    return data;
  },

  async listar(): Promise<ClienteResponseDto[]> {
    const { data } = await apiClient.get<ClienteResponseDto[]>('/clientes');
    return data;
  },

  async eliminar(id: number): Promise<ApiResponseDto<ClienteResponseDto>> {
    const { data } = await apiClient.delete<ApiResponseDto<ClienteResponseDto>>(
      `/clientes/eliminar/${id}`,
    );
    return data;
  },
};
